### DISCUSSION TITLE
BRouter profile settings in Cruiser

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#138] [Q&A] [answered]
BRouter **avoidance** options are already available in the latest Cruiser:
- https://github.com/devemux86/cruiser/discussions/134

We can see how we can offer more BRouter profile options in Cruiser.

For example there is the "**vmax**" (target speed) profile parameter,
which also changes the roads used in BRouter route calculations.

Do you think some other BRouter profile options would be useful?

---

BRouter profile options were also explored in the past by other users:
- https://forum.kurviger.de/t/app-offline-routing-brouter/649/19

You can learn and play with the profile options on the [BRouter website](https://brouter.de/brouter-web/).

**Comment by DolomitesTom** [comment]
> Do you think some other BRouter profile options would be useful?

yes, for sure.

**Comment by Ciutnik** [comment]
In my opinion, everything that can make the routing algorithm similar to the Kurviger is an argument for switching from it to the Cruiser. I am closely watching the development of both applications, because the situation in which I set routes in one and navigate in the other is very uncomfortable.

I love Cruiser's UI and logic, but the inability (or just only my inability?) to route secondary and tertiary local roads keeps me from paying the subscription fee (which I probably will sooner or later anyway) :)

**Reply by DolomitesTom** [reply]
Everyone can decide that for themselves, but I still find it a little petty to make such considerations because of €10 a year for a great product.

I also plan larger tours with Kurviger Tourer with a little more shaping points, export, import, done. Seriously, what's uncomfortable about that?

**Reply by devemux86** [reply]
I will probably add a separate section for the _BRouter profile options_.
They are extremely powerful and we will include many more of them!

**Reply by Ciutnik** [reply]
@DolomitesTom do not get me wrong. I'm not talking about 10 Euro, but about convenience. I agree that when planning a large trip, it is not burdensome to use two applications or an application and a website. But when planning an ad hoc half-day trip that you'll change X times along the way, you want to make it as quick and simple as possible.

**Reply by DolomitesTom** [reply]
Sure, the lack of Kurviger routing was also a pain for me in the beginning. But with Cruiser and Brouter everything works without any internet connection. And this is a benefit in many Alpes regions and also other regions in Germany.

**Comment by WolfgangZ42** [comment]
> For example there is the "vmax" (target speed) profile parameter

I'm using two profiles (Moto.brf / Auto.brf), based on the "old" Kurviger-Profiles, and the difference between them is only **avoid motorways** and **vmax**. 
<img width="800" src="https://user-images.githubusercontent.com/121806238/235298709-f82e0339-583d-40b2-9a01-6cf31099168a.jpg"/>
The first one can be triggered via Cruiser pref file already - would be great if the latter one could also be specified. Then the original **car-fast.brf** could be use without any changes. Even the nogo-vetos can be assigned to motorcar_short only (for my Munich problem ...).

**Reply by devemux86** [reply]
> would be great if the latter one could also be specified

It is already in my implementation list. :slightly_smiling_face: 

What min / max limits should we use for the "vmax" option?

**Reply by WolfgangZ42** [reply]
For my opinion 60 to 160 should be sufficient.

**Reply by devemux86** [reply]
"vmax" is related to the speed stored in the OpenStreetMap database?
So BRouter can choose the appropriate roads in the routing.

(not the speed we drive)

**Reply by Ciutnik** [reply]
I noticed that the vmax value of 60 km/h is kind of an important breaking point in routing. This means that when I set 60 km/h, primary roads are heavily used for routing; when set to 59 km/h, primary roads are avoided (which I like). Therefore, I propose to give the option to choose this vmax value from 50 km/h.
Of course, I mean the value set in the file *.brf

**Reply by devemux86** [reply]
> Therefore, I propose to give the option to choose this vmax value from 50 km/h.

No problem, we can set any min / max limits for the vmax we want.
BRouter website does not seem to have min / max limits in its interface, so we cannot use its values.

The vmax should be related to a specific OpenStreetMap speed tag?

**Reply by WolfgangZ42** [reply]
As I'm learning at best by "trial and error" I did use two BRouter profiles, one with vmax=70 and another with vmax=50 to see the difference. It's not sooo clearly as you may see in this screenshot:
<img width="600" src="https://user-images.githubusercontent.com/121806238/236609082-9aefcedd-790b-4a9b-8d2e-2fcc05861d54.jpg"/>  
The Kurviger and Calimoto routes are really different ...

**Reply by devemux86** [reply]
@WolfgangZ42 what happens in your example if use BRouter vmax **+** avoid options?

**Reply by devemux86** [reply]
@WolfgangZ42 

What profiles are in Calimoto and Kurviger?

Are their "fastest" profiles really fastest? They still seem to avoid some fast roads.
Kurviger "fastest" route description: **"Mostly"** motorways and major highways.

**Reply by WolfgangZ42** [reply]
> BRouter vmax + avoid options?

Didn't use any avoid option in Cruiser, but in each BRouter profile there is avoid_motorways set as "true".  

> What profiles are in Calimoto and Kurviger?

It is a test of a motobike tour, so in each website was "curvy" set. 

For forensic investigation here all tracks: [AR_tracks.zip](https://github.com/devemux86/cruiser/files/11411846/AR_tracks.zip)

**Reply by Ciutnik** [reply]
> I did use two BRouter profiles, one with vmax=70 and another with vmax=50

I tried the same comparison (on BRouter-web) and the difference was significant. While 70 km/h is on an expressway, 50 km/h runs mainly on local tertiary roads. So it depends on the case. I didn't change anything except the vmax value.
Edit: The highway avoidance option has no effect on the expressway used in this example!

![50_vs_70](https://user-images.githubusercontent.com/122647782/236613608-8dad93dd-6126-4619-b970-b13000066e02.png)

**Comment by Ciutnik** [comment]
If, like me, you definitely want to avoid both motorways (for which there is now a switch) and main roads, some solution is the trick of forcing very low speed on these road categories during routing. I used this successfully several years ago using another routing application. I just checked that with BRouter ~~(for now in the web version)~~ it also gives me satisfying results!

Well, by replacing the vavalues e.g. with the following:
```
assign maxspeed_implicit =
  switch highway=motorway                   30
  switch highway=motorway_link              30
  switch highway=trunk                      30
  switch highway=trunk_link                 30
  switch highway=primary|primary_link       30
```
you can reach your destination without the danger of completely blocking the main road in the event that there is no other local road nearby. It may not be a perfect solution, but it is very effective. Somehow I missed it before.

What's more! Modifying in a similar way section `assign maxspeed_explicit =` you can you can limit the passage through cities through `switch maxspeed=urban 30`.

**Reply by devemux86** [reply]
@Ciutnik thanks for the extra parameters!

I will check them and how they can be called from the Cruiser environment.

**Reply by Ciutnik** [reply]
Perhaps on the basis of the above-mentioned speed changes in the brf profile, you can do something like setting the user preferences of the roads used. Once again, I will refer to my good old Garmin 276c, which, in addition to choosing the 'Fast' / 'Short' route option, had 'Own user', realized by preference sliders of different road categories. Something like this:
```
Category     Preference

Expressway   [#.........]  1
Main         [..#.......]  3
Secondary    [........#.]  9
Tertiary     [......#...]  7
```

**Comment by airald** [comment]
I'm used to plan routes using waypoints to force to use the ways I want. The problems are sometimes with certain roads that have a lower category as terciary roads and are not transitable with standard car profiles, even they are allowed to all but are highway = track. I haven't found it yet the parameter to be able to use this kind of ways. If I use a bicycle profile, Brouter use this kind of tracks all the time and it's impossible to plan a route. Brouter is very powerfull, but I need to investigate and learn a lot about it.

**Comment by devemux86** [comment]
BRouter separate options / profile will be available in the next Cruiser:

<img src="https://user-images.githubusercontent.com/3484020/236611323-2d931765-a6b1-48c5-9c43-4b1eab20a25a.jpg" width="33%"/><img src="https://user-images.githubusercontent.com/3484020/236838923-ff339aeb-ee41-4e3c-bfbf-9302924be75c.jpg" width="33%"/><img src="https://user-images.githubusercontent.com/3484020/236611331-f9a2d55e-2b8e-49f0-8778-15cb056f6dce.jpg" width="33%"/>

**Comment by Ciutnik** [comment]
BRouter-web has the option to choose between a "Primary" route and three "Alternative" routes. I have noticed that these alternatives can be extremely interesting. It would be great if Cruiser also allowed you to make this choice. It sounds so tempting that it's probably impossible?

**Reply by devemux86** [reply]
@Ciutnik thanks for the suggestion.

BRouter does not include the alternative routes in the same response,
but it requires separate route requests and use the `alternativeidx`.

It becomes more complicated for an app to request and receive everything in one call.
But it should be possible, I will check it!

**Reply by Ciutnik** [reply]
I'm guessing "Alternatives" are only available for BRouter online? I think the easiest minimum option would be to set the vehicle properties to use a user-selected alternative number. But of course, it would be best to be able to select one of the four options (not necessarily presented simultaneously) before navigating...

**Comment by devemux86** [comment]
Alternative routes are also perfectly possible with BRouter (offline) and will be available in the next Cruiser.

**BRouter has the advantage of being able to calculate alternative routes + having via / shaping points.**
Most routers require only 2 waypoints (start + destination) to calculate alternative routes.

Here is an example of BRouter (offline) with via points and 4 alternative routes!
Route selection is done by pressing the route on the map.

<img src="https://user-images.githubusercontent.com/3484020/236498184-9523fae8-31f8-4640-bfc6-84208748c9df.jpg" width="50%"/><img src="https://user-images.githubusercontent.com/3484020/236499762-b99714d3-df96-41f7-8252-889432f4e6bc.jpg" width="50%"/>

**Reply by Ciutnik** [reply]
Wow! This can be a real game changer! Thanks a lot!

**Reply by devemux86** [reply]
@Ciutnik thanks for the reminder!

It was already in my list for a long time. :slightly_smiling_face:

**Reply by devemux86** [reply]
BRouter options will have their own more detailed "**Alternative routes**" setting to select their number.

See the updated images above:
- https://github.com/devemux86/cruiser/discussions/138#discussioncomment-5815828

**Reply by 0709wiwiwi** [reply]
And so can even be saved and shared this for later use all this into a single gpx navtrack file.   
The above example can already be shared this to make it even more tempting at this stage.

**Reply by devemux86** [reply]
> And so can even be saved and shared this for later use all this into a single gpx navtrack file.

@0709wiwiwi interestingly, alternative routes were not enabled in the GPX navigation track export.

It will be enabled in the next Cruiser update.

**Reply by devemux86** [reply]
@0709wiwiwi attached the exported GPX: [4_routes.gpx.zip](https://github.com/devemux86/cruiser/files/11408394/4_routes.gpx.zip)

**Reply by 0709wiwiwi** [reply]
You need to separate the 4 navtracks by the tag trk.
See example attachment: 4_navtrk_Cruiser.gpx
[4_navtrk_Cruiser.gpx.txt](https://github.com/devemux86/cruiser/files/11411881/4_navtrk_Cruiser.gpx.txt)
The 4 different (selectable) track displays by Garmin Mapsource.
![image](https://user-images.githubusercontent.com/54167201/236618161-db384a13-2f80-4382-a18e-5bfd0b956eff.png)


**Note.**
_A single track can be composed of several trkseg.
The trkseg are invisibly connected to each other. (OK)
Some soft connect trkseg visibly anyway. (NOK)
(Here shown as a single track display by Garmin Mapsource)
![trkseg visually connected false](https://user-images.githubusercontent.com/54167201/236617953-92b47025-8c10-49cd-bf18-7e330d3365e3.png)
Result is then a sloppy and confusing display._

[Connection line between gpx trkseg or not.txt](https://github.com/devemux86/cruiser/files/11411887/Connection.line.between.gpx.trkseg.or.not.txt)

**Reply by 0709wiwiwi** [reply]
What follows is unaddressed and so is not tested. 
I don't even know if you are interested in the feature.
![image](https://user-images.githubusercontent.com/54167201/236620676-897ac9d2-d834-4174-8af8-0599afa94091.png)
[4_navtrk_color_Cruiser.gpx.txt](https://github.com/devemux86/cruiser/files/11411956/4_navtrk_color_Cruiser.gpx.txt)
Open the attached gpx file (text) and note the content .

**Reply by devemux86** [reply]
Similarly to [favorite groups](https://github.com/devemux86/cruiser/discussions/45#discussioncomment-5791140), better to avoid any custom gpx extensions in export?
These can be better handled in the app after import with more customizations.

Favorite routes and tracks will arrive soon with their own manager in Cruiser:
- https://github.com/devemux86/cruiser/discussions/35
- https://github.com/devemux86/cruiser/discussions/81

**Reply by 0709wiwiwi** [reply]
> better to avoid any custom gpx extensions in export?

The main thing is that all necessary basic functions are placed in default gpx tags. 
Any other apps can directly read if they need or want too participate.
The limited number of tags available in the gpx standard should not stop you from offering those slightly less important but ooh very nice extras to be offered by the Cruiser app. By the way, not all apps have to be or even become that much fun don't they ?

**Reply by devemux86** [reply]
> You need to separate the 4 navtracks by the tag trk.

@0709wiwiwi thanks for the hint!

I updated the export, attached the exported GPX: [4_routes.gpx.zip](https://github.com/devemux86/cruiser/files/11412179/4_routes.gpx.zip)

**Reply by devemux86** [reply]
> The main thing is that all necessary basic functions are placed in default gpx tags.
Any other apps can directly read if they need or want too participate.

@0709wiwiwi indeed compatibility is most important.

Let's wait for next Cruiser release with the favorite groups with icons + colors,
see how users use them and then can discuss if the export needs changes.

**Reply by WolfgangZ42** [reply]
> Route selection is done by pressing the route on the map or in instructions list.

Will be all alternate routes visible while navigation? And could you imagine that a switch between these routes can be done "in the flight"? Sure it's possible to stop navigation and start it again with an alternate route, I think.

**Reply by devemux86** [reply]
> Will be all alternate routes visible while navigation?

Navigation renders the active route (normal or advanced appearance).
We can change route like this:
- Pause navigation (all routes appear)
- Select alternative route (tap route)
- Resume navigation

**Reply by Ciutnik** [reply]
@devemux86 One minor request about alternative routes: gray color is OK, but could their **transparency** be inherited from the color of primary route? (probably not worth doing a separate setting). It would make it easier for me to view them. Ah, force of habit 🙏

**Reply by devemux86** [reply]
@Ciutnik 

Good point! :+1: 

And with BRouter supporting multiple alternative routes, we probably need different (gray) colors?

**Reply by Ciutnik** [reply]
> we probably need different (gray) colors

I don't think so. For purity, they can all be the same gray. And so we choose the one we are interested in with our finger on the screen, so they probably do not have to be differentiated so as not to be confused with the main active route.

**Reply by devemux86** [reply]
> One minor request about alternative routes: gray color is OK, but could their transparency be inherited from the color of primary route?

It will be available in the next Cruiser update. :slightly_smiling_face:

**Reply by Ciutnik** [reply]
Great! Thank you very much! 

By the way, now that we're on the subject, wouldn't it be nice to see other alternate routes permanently during navigation one of the selected ones? Is it even possible? Sometimes you want to spontaneously change your route in the middle of your journey because there is a better road somewhere or a nicer view. In that case, it's nice to see alternatives that certainly leads to the planned destination. These would be such ready-made ready-to-use routes.

**Reply by devemux86** [reply]
Now we do not show the alternative routes inside navigation to _keep the display clear with our route_.
Also there is the advanced route appearance in navigation that: "Show only the active route segment".

Nevertheless, I could show the alternative routes in navigation, if you need it?
What about the advanced route appearance?

Currently we can "Pause" navigation > select an alternative route > "Resume" navigation.

**Reply by WolfgangZ42** [reply]
> keep the display clear with our route.

Thats right, but: I'd like to try what happens if alternate routes are shown. The advanced route appearance is top and should be kept. 

> Currently we can "Pause" navigation > select an alternative route > "Resume" navigation.

Fair enough for switching to an alternate route ...

**Reply by Ciutnik** [reply]
>> Currently we can "Pause" navigation > select an alternative route > "Resume" navigation.

> Fair enough for switching to an alternate route ...

Not exactly. You would have to remember the course of the alternative route and be able to visualize it in relation to the active route. I mean a situation where you need to make a quick decision when you hit a road you don't want to go (bad surface road). When you see an alternative route, you just glance at the screen and you know what to do without fear of "getting lost".

This is a solution for those who do not like automatic recalculation.

<img src="https://github.com/devemux86/cruiser/assets/122647782/eac606f4-64a4-48ad-b153-5898795db5ee" width="50%"/>

**Comment by Ciutnik** [comment]
> It was already in my list for a long time

This explains how you were able to implement such complex functionality in just 10 minutes 😃👍

**Reply by devemux86** [reply]
> This explains how you were able to implement such complex functionality in just 10 minutes 😃👍

I implemented it for [CycleStreets](https://www.cyclestreets.net/) bike routing that calculates 3 routes (Balanced, Quietest, Fastest)
so was ready to enable it for BRouter as well. :slightly_smiling_face: 

<img src="https://user-images.githubusercontent.com/3484020/236505149-2355626e-7091-47c3-bdbe-464d222dac12.jpg" width="50%"/>

**Comment by abdullahO2** [comment]
Thank you very much for these important improvements

**Comment by devemux86** [comment]
@DolomitesTom did you solve your report?
(it is not here anymore)

It is best to avoid changing at the same time:
- BRouter profiles externally
- BRouter settings in Cruiser

I tested with the original BRouter profiles and BRouter avoidance settings seem to work?

**Reply by DolomitesTom** [reply]
yes, I sorted it out. The mapping in Brouter also has to be correct for any profile, car-fast, car-eco and others. If this not maps to the same settings as in Cruiser, then it doesn't work as expected.

**Comment by devemux86** [comment]
BRouter options for **bike** will also have their own location in Cruiser settings.
Do you have specific BRouter bike options you want included?

You can check all of them in [BRouter website](https://brouter.de/brouter-web/).

<img src="https://user-images.githubusercontent.com/3484020/236631825-7c7e3e56-d680-4146-ac7a-118fc0b9bc94.jpg" width="33%"/>

**Comment by MotoUKRider** [comment]
Can you "handle" the `barrier` and `gate` permissions in here as per https://github.com/devemux86/cruiser/discussions/56 ?

**Reply by devemux86** [reply]
> Can you "handle" the barrier and gate permissions in here as per https://github.com/devemux86/cruiser/discussions/56 ?

I will check it!

**Reply by devemux86** [reply]
We need to see if the "assign" command can be used externally.
Also mentioned for another use above:

- https://github.com/devemux86/cruiser/discussions/138#discussioncomment-5761524

**Comment by devemux86** [comment]
I updated above the **bike** profile options that will be included:
(similar to BRouter website)

- https://github.com/devemux86/cruiser/discussions/138#discussioncomment-5822895

**Comment by DolomitesTom** [comment]
> Do you have specific BRouter bike options you want included?

the problem is I am not familar with the Brouter settings. Is there a proper documentation, maybe in german, how this all works together?

**Reply by devemux86** [reply]
There is a BRouter profile guide (don't know about other languages):
- https://github.com/abrensch/brouter/blob/master/docs/features/costfunctions.md
- https://brouter.de/brouter/costfunctions.html

**Reply by DolomitesTom** [reply]
Thanks will read it when having time.

**Comment by devemux86** [comment]
Another improvement is to allow users to easily **select the BRouter profile inside Cruiser**:
(if you want more profiles please let me know)

<img src="https://user-images.githubusercontent.com/3484020/236838923-ff339aeb-ee41-4e3c-bfbf-9302924be75c.jpg" width="50%"/><img src="https://user-images.githubusercontent.com/3484020/236838932-318d68af-f963-4910-ac5b-501f8b2ca16d.jpg" width="50%"/>

**Comment by Ciutnik** [comment]
> select the BRouter profile inside Cruiser

Looks promising! Will it be possible in the future to divide the route into sections and assign a different profile to each section? I want to be able to drive to the city border using the "fast" profile and then go around the area using the "curvy" (economic) profile. So what we like the most 🙂

**Reply by devemux86** [reply]
> Will it be possible in the future to divide the route into sections and assign a different profile to each section?

This is already in my list for implementation, please see here: :slightly_smiling_face: 

- https://github.com/devemux86/cruiser/discussions/142

**Reply by Ciutnik** [reply]
Great! Thank you very much! I was sure I read an announcement of this functionality somewhere, but I couldn't find where.

If I understand correctly, BRouter profiles will still be editable by the user, because I have very good experience with this (I shared my thoughts above).

**Reply by devemux86** [reply]
> If I understand correctly, BRouter profiles will still be editable by the user, because I have very good experience with this (I shared my thoughts above).

Current and next Cruiser use the profiles in BRouter app.

I will check the idea mentioned in the barriers / gates discussion:
(if successful, these BRouter profiles should also be editable)

- https://github.com/devemux86/cruiser/discussions/56#discussioncomment-5839071

**Comment by devemux86** [comment, marked as answer]
BRouter profile selection and settings are available inside [Cruiser 3.0.19](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

"Settings | Routing | BRouter" or search settings for "brouter".

**Reply by DolomitesTom** [reply]
is there a difference between avoid motorways in Brouter and avoid "Schnellstraßen" in Options? Does "Schnellstraßen" mean motorways or federal highways?

**Reply by devemux86** [reply]
BRouter now works exclusively with its own settings inside Cruiser in "Settings | Routing | BRouter".

The "Settings | Routing | Options" do not affect the BRouter route calculations.
They are useful for all other routing services.

**Reply by DolomitesTom** [reply]
thanks good to know.

**Comment by 0709wiwiwi** [comment]
Multiple choice. The chances of getting it right immediately diminish.
![image](https://github.com/devemux86/cruiser/assets/54167201/d569a2b5-741d-4712-827b-b47411e478f6)

**Reply by devemux86** [reply]
@0709wiwiwi I am not sure what do you mean?

They are different concepts from separate apps:
- **Cruiser Navigation profile**: select travel mode (car, motorcycle, bike, walk)
It works with all routers and affects navigation distances, auto zoom, route duration, ...

- **BRouter profile**: select the BRouter routing mode
It has its own names and can be combined with any travel mode.

You can now select the BRouter profile inside Cruiser, instead of selecting it externally in the BRouter app.

**Reply by 0709wiwiwi** [reply]
Many options and a simple menu becomes more difficult.  
With the v 3.019 I surpringly had to puzzle right away. 
You need to master both Brouter and Cruiser menu for this. 
(What has been put in the Brouter app must match what follows).
Certainly this combination is not that obvious for newbies.
In the routing menu see Bike at the top. 
The one on the top is the boss or maybe he is not ?
Proceed to Brouter, Car+motorcycle menu is present.
 
In the router menu, maybe you should start by the Service ? 
Then choose the means of transport that fits the router engine? 
Other KISS ideas certainly welcome.

**Reply by devemux86** [reply]
I can make the BRouter profile to be auto-selected when navigation profile is changed.
(users can always change again the BRouter profile)

**Comment by MotoUKRider** [comment]
If you use your own modified BRouter profile (like mine, without gates/barriers), you can't select it from Cruiser.  I had to rename `motorideruk.brf` to `car-fast.brf`.

I also found the duplicated "avoid" options a bit confusing.

**Reply by devemux86** [reply]
> If you use your own modified BRouter profile (like mine, without gates/barriers), you can't select it from Cruiser. I had to rename motorideruk.brf to car-fast.brf.

You are right, now we call directly the BRouter profiles.
We need to improve the BRouter experience in Cruiser.

Include BRouter profiles in Cruiser (with extra options + allow modifications) is an interesting idea:
- https://github.com/devemux86/cruiser/discussions/56#discussioncomment-5839071

The thing is that most users are not familiar with custom profiles or the mapping of BRouter modes:
(bicycle | foot | motorcar + fast | short combinations)

We need to make the user experience easier while keeping the flexibility of BRouter available.

<img src="https://github.com/devemux86/cruiser/assets/3484020/22c0c1fb-0c26-416e-867b-75a55b808af8" width="33%"/>

**Reply by MotoUKRider** [reply]
Being able to pick a custom BRouter profile within Cruiser is fine - providing it can access the actual folder of `.brf` files :)  My custom profile is bound be useful for others (pesky level-crossings!) so if it can be included within Cruiser for new installs it has my blessing.  The issue of a new user saying: "well, my other app dealt with this route OK." shouldn't occur :)

It seems that there are a number of riders here who rely on different routing options, eg: avoid tolls/motorways/borders/unpaved, very curvy, ride in a westerly loop for 120km, announce turns 123.775m away, keep me travelling at 60kmph etc.  User-friendly configuration, away from text files if possible, would be the ideal solution.  "Power Users" could make their own edits if they wanted.

I'm only using BRouter because I can tweak it to work, and have some programming knowledge.  If Graphhopper behaved itself on the back-roads then I'd probably have stuck with that :)

**Reply by devemux86** [reply]
> User-friendly configuration, away from text files if possible, would be the ideal solution. "Power Users" could make their own edits if they wanted.

:+1: 
This is exactly the purpose behind BRouter's latest changes.
As much as we can satisfy both groups of users.

**Comment by Leeb64** [comment]
The routing menu is extremely confusing for non-CRUISER users. I would almost say deterrent (my wife doesn't check it).
I think you have to first select the routing service, then the profile, then options.
The whole thing has to be "idiot proof" somehow.
I know: not easy...

**Reply by devemux86** [reply]
I agree with all of you, we need to simplify these settings somehow!

**Reply by 0709wiwiwi** [reply]
@Leeb64 

> To first select the routing service, then the profile, then options.

I hope that [this routing panel](https://github.com/devemux86/cruiser/discussions/193#discussioncomment-6056986) proposal allows and so encourages your named menu 123 sequence.
What do you think ?

**Reply by devemux86** [reply]
The routing menu is already full of buttons, with probably 1 empty space.
We should be careful, otherwise it can be inconvenient: https://github.com/devemux86/cruiser/discussions/82#discussioncomment-6073648

To host more buttons, we would have to make their size smaller.

**Reply by 0709wiwiwi** [reply]
I see no need to add any more buttons. Full is full indeed.  However, in my opinion it is very fine finished like this now.
_The History button eventually is removable if a usefull on map ↶ ↷ Undo Redo function **at routeplanning** is available._
I couldn't think of anything fine 3 weeks ago then either.  Sometimes it's best to let something take a while to mature.
With @.... I also ask first for the opinion of those who have already contributed to this topic, of course. 
Have a nice evening all !

**Reply by 0709wiwiwi** [reply]
@ Emux.
An updated example you may find more favourable. 

![Main Panel to Routing Panel](https://github.com/devemux86/cruiser/assets/54167201/a923bed1-8380-4798-a73d-f976bb1e26b0)

More suitable for both Portrait & Landscape this way.

**History Button.** 
The faster it can disappear the better.
History is useful but the Button application is clumsy.
If you want to go back a few steps you mostly choose the wrong time.  
An on map [undo redo](https://github.com/devemux86/cruiser/discussions/199) function quickly shows you all previous states. 
Superiorly useful.

**Reply by devemux86** [reply]
Undo in route planning is already in my list. :slightly_smiling_face: 

Question: how often do we change the router and we need a button in the front menu?

**Reply by 0709wiwiwi** [reply]
Since you provide a lot of route planners in Cruiser, it is desirable to be able to select them quickly as well.
It has also been asked and is implemented to present the active routing service clearly informative.
It makes little sense when you offer so many services that these are too far away in the menu to set.

Sure the offline routing services become the most valuable.  Are so many online planners even necessary ?  
**Mapsforge maps with accompanying automatic GH routing download is especially attractive for newbies.**
If something does not immediately functions to their liking, this is quickly followed by a de-installation.  
These instantly so lost new users, try to avoid this by an immediate powerfull ready to use map install offer.
The Brouter app is flexible, but the fact you first have to download the app and routing data puts off new users. 
Only once they have than a bit more Cruiser app experience will this additional step than be taken and enjoyed.

**Reply by 0709wiwiwi** [reply]
> we need a button in the front menu?

Should you ever offer to mix routing services.
A smooth selection of Services is expected.

- Services mixing is possible ? 

A navtrack design by Cruiser_osrm online. 
Start navigation in off route position.
Cruiser prompts you towards the Start ? 
The connection is by Cruiser_BRouter offline.
Save that new result than as a navtrack.

[Cruiser_BRouter_OSRM_20230603_115020.zip](https://github.com/devemux86/cruiser/files/11642374/Cruiser_BRouter_OSRM_20230603_115020.zip)

Navtrack BRouter offline and osrm online.

**Reply by devemux86** [reply]
> Should you ever offer to mix routing services.

Currently it is available in navigation with rerouting.

It will also become available in route planning:

- https://github.com/devemux86/cruiser/discussions/142

**Comment by WolfgangZ42** [comment]
Yesterday I was surprised by the BRouter extension in Cruiser - could not define a destination (okay - I removed all default profiles from PROFILE2): 
![2023-05-12 08_55_30-2023-05-11 11 25 16 png - IrfanView (Zoom_ 470 x 939)](https://github.com/devemux86/cruiser/assets/121806238/6976dc0b-9825-4a7c-881b-b088f4bcbf64)

Not a big issue for me now, but when I imagine being on a 3 day tour with friends and not having a fallback like Kurviger 2 ... 

Of course I can copy my profiles (#Auto > car-fast, #Moto > car-eco). Then at the next BRouter update this has to be done again. Defining own profile names in Cruiser would help here. Default should be the current BRouter profiles of course. An own profile directory in Cruiser could also be a solution (see https://github.com/devemux86/cruiser/discussions/161#discussioncomment-5876202).   
My particular problem (Munich, you know) then is the exclusion of "nogo-vetos" - is this controllable by Cruiser at all? 
In /settings/routing IMO the "_Route type_" should be presented directly below the "_Navigation profile_" (instead of in /settings/routing/options). And could you imagine to add a switch in /settings/routing/BRouter to totally switch off the Cruiser BRouter profile setting? For those who prefer their own setup? 
Since I plan and perform my bike tours and walks with Komoot, I do not need these profiles in Cruiser. So far I have used them for experimental profiles also for motorcycle tours: 
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/1e25d8ca-6b2c-4c9d-bd8f-9ba781019480"/>

Of course this should not be considered in the BRouter settings of Cruiser, especially since very good options are already offered in /settings/routing/BRouter/Bike.

**Reply by devemux86** [reply]
@WolfgangZ42 thanks for the ideas!

The whole implementation will definitely be improved, it's just the 1st step to a closer integration with BRouter.

Currently in your profiles you need to use the BRouter original file names, e.g. `car-fast.brf`.
Does it help for now?

See also the similar comment above by @MotoUKRider:
- https://github.com/devemux86/cruiser/discussions/138#discussioncomment-5870981

**Reply by WolfgangZ42** [reply]
Sure - I've already copied the .brf files. And the nogo vetos are further suppressed as established in the BRouter App. Everything works! 

A new line is shown by BRouter and a tap on it offers information about the actual route: 
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/23578142-cc54-455f-b6c4-9ff2926d859d"/>   <img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/5338b3c7-b67b-4eed-9fca-86ff6e00cf17"/>

**Comment by Leeb64** [comment]
The "car-fast" and "car-eco" profiles are identical as far as I know. They only differ in the target speed. But that can be freely selected in the CRUISER....
Kind of doubled..

**Reply by devemux86** [reply]
@Leeb64 thanks, you are right!
Another thing to improve. :slightly_smiling_face:

**Reply by DolomitesTom** [reply]
and the car-fast and car-eco are overwritten if only segments are downloaded. That's bad because all changes are lost. Then I can always overwrite it again from a backup copy.

**Reply by devemux86** [reply]
> and the car-fast and car-eco are overwritten if only segments are downloaded

Do you mean in BRouter media folder?
You do not reinstall BRouter, only use its download manager?

This should not happen, can you report it in [BRouter repository](https://github.com/abrensch/brouter/)?

**Reply by afischerdev** [reply]
Did you read the readme.txt in profile folder?
`Please copy your profile before you edit.`
Standard profiles will be overwritten when new. To make sure you always have the latest ones.

**Reply by devemux86** [reply]
@afischerdev thanks for the explanation!

The confusion started with the new Cruiser that uses directly the standard profile names in BRouter routing.
Previous Cruiser used the BRouter profile mapping, though it seems more complicated for no power users.

I am already looking how we can improve this situation.

**Comment by DolomitesTom** [comment]
I guess we need more Brouter specific parameters in Cruiser in order to not needing to edit the original brf files or even rename it, for example like this urban 30 proposal in order to not route through cities or mabe also some new in order to realize curvy roads. If you are working together with afischer what seems to be a good connection, maybe this is somewhen possible 🙂.

**Comment by Cer-o** [comment]
Hello, everyone. I have read the entire thread and due to my limited knowledge and the difficulty of doing it through a translator, I think I will have understood about 30% of what has been said (which for someone with my little experience is a lot).

I would like to ask two very basic questions:
1.- Is BRouter in charge of designing the route between the beginning and the end of the route that I mark in Cruiser?
2.- If so, what parameters should I configure to choose between straight paths, curves or many curves?

Thank you very much and sorry for the inconvenience.

**Reply by devemux86** [reply]
@Cer-o please ask anything you want to know, we are all here to help!
You can also write in your language. :slightly_smiling_face: 

1. The re-routing in Cruiser does not change the routes.
(it only calculates until the next waypoint)

3. To avoid main roads and use small back roads, please see the BRouter settings in the image below:
(including the speed setting)

<img src="https://github.com/devemux86/cruiser/assets/3484020/153ae992-b337-46a4-8a43-6a3993f0799a" width="50%"/>
